packages/launchdarkly-adapter/modules/adapter/adapter.js

```diff
@@ -34,23 +65,30 @@ const subscribeToFlagsChanges = ({ rawFlags, client, onFlagsStateChange }) => {
 };
 
 // NOTE: Exported for testing only
-export const createAnonymousUserKey = () =>
+export const createAnonymousUserKey = (): string =>
   Math.random()
     .toString(36)
     .substring(2);
 
-const ensureUser = user => ({
+const ensureUser = (user: User): User => ({
   key: user && user.key ? user.key : createAnonymousUserKey(),
   ...user,
 });
-const initializeUserContext = (clientSideId, user) =>
+const initializeClient = (clientSideId: string, user: User): Client =>
   initialize(clientSideId, user);
-const changeUserContext = (client, nextUser) => client.identify(nextUser);
+const changeUserContext = (nextUser: User): void =>
+  adapterState.client && adapterState.client.identify
+    ? adapterState.client.identify(nextUser)
+    : undefined;
 
 // NOTE: Exported for testing only
-export const camelCaseFlags = rawFlags =>
-  Object.entries(rawFlags).reduce((camelCasedFlags, [flagName, flagValue]) => {
-    const [normalzedFlagName, normalzedFlagValue] = normalizeFlag(
+function entries<T>(obj: { [string]: T }): Array<[string, T]> {
```

Comment:
The exported comment is for the `camelCaseFlags` function right?

Reply:
Curious why have you "extracted" this function

Reply:
My friend. You don't want to know. This cost me about 90 minutes and got me mad on a 🍷. It's due to the fact that flow seems to loose type information through the `Object.entries`. Reference on their repo [here](https://github.com/facebook/flow/issues/2174).

Reply:
I know one thing or two about those "hours" 😅

Reply:
Maddening. That's the only word coming to mind. However, there is little I honour more than all the hours people spend in building these great libraries. 👍